### PR TITLE
Ft reset password

### PR DESCRIPTION
**Pull request by GilbertUp**
## What does this PR do?
-It allows users to reset their password by letting them provide their email to receive password reset links and allows users to reset passwords by entering a new password and confirming it.

## How should this be tested?
-Clone this [repo](https://github.com/atlp-rwanda/Dominators-Phantom-FE/) 
-checkout to the branch name  #ft-reset-password
-Run the command 
-run the command npm i 
-run the command npm start to start a server
-Click on forgot password
-enter your email
-Send link
-To access the password reset form input its  route in the URL field (http://localhost:3000/reset-pwd) this will be accessed after receiving the password reset link in the user's email and it will be done by backend-team
## What to expect?
-if the email is in invalid format or missing, an appropriate message will be displayed
-if the password is missing or less than 8 characters, an appropriate message will be displayed
-if the password mismatches and if it does not contain a number, an appropriate message will be displayed

### CODE REVIEW

**Review by aimelive — APPROVED** [review]

**Review by UmuhireJessie — APPROVED** [review]

**Comment by Anguandia** [comment]
Please rebase and squash commits as we review and test locally. This PR is long overdue, treat is as a matter of urgency, @GilbertUp

File: src/components/ResetPassword/ResetPassword.css [hunk 1]
```diff
@@ -1,209 +1,177 @@
...
+  /* position: absolute; */
+  width: 500px;
+  height: 40px;
+  background: #eff2f3;
+  border: 2px solid #c5c4c4;
+  box-sizing: border-box;
+  padding: 5px;
+  display: block;
+  justify-content: space-between;
+}
```
Comment (furebo) [inline src/components/ResetPassword/ResetPassword.css]: Please add the margin auto  to this elements.

File: src/components/ResetPassword/ResetPwdComponent.js [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
  const[confirmPassword,setConfirmPassword] = useState("")
- const redirect = useNavigate();
+
  const handleSubmit = (e)=>{
   e.preventDefault();
   if(validatePassword()){
     if(password===confirmPassword){
       toast.info("password reset successfull!")
-      redirect("/login")
     }
```
Comment (furebo) [inline src/components/ResetPassword/ResetPwdComponent.js]: Integrate with Backend to reset password before we get the successful message

**Review by furebo — CHANGES_REQUESTED** [review]
Please Also remember the good description of the PR

File: src/App.js [hunk 3]
```diff
@@ -27,10 +27,9 @@ function App() {
       <Route path="/users" element={<DriverOperator />} />
       <Route path="/roles" element={<Roles />} />
       <Route path="/Login" element={<Login />} />
-      <Route path="/Reset-link" element={<ResetLink/>} />
-      <Route path="/Reset-pwd" element={<ResetPwd/>} />
-      <Route path="/Dashboard" element={<Dashboard/>} />
-      
+      <Route path="/Reset-link" element={<ResetLink />} />
+      <Route path="/Reset-pwd" element={<ResetPwd />} />
```
Comment (Anguandia) [inline src/App.js]: @gerard1001 remember to change these paths to reset

**Review by Anguandia — APPROVED** [review]
@gerard1001 , I thought this should be ready by now, please note:
- modify the PR description, especially how to test
- I hope we are using real users from the database, not the users.json
- ping @furebo after pushing so he can resolve his change request.

**Review by Anguandia — CHANGES_REQUESTED** [review]
Remove any redundant files/pieces of code

**Review by Anguandia — APPROVED** [review]

**Review by furebo — APPROVED** [review]

**Pull request merged**